marathon_acme/tests/test_clients.py

```diff
@@ -322,6 +350,85 @@ def test_url_overrides(self):
         request.setResponseCode(200)
         request.finish()
 
+    @inlineCallbacks
+    def test_json_content(self):
+        """
+        When a request is made with the json_content callback and the
+        'application/json' content type is set in the response headers then the
+        JSON should be successfully parsed.
+        """
+        d = self.cleanup_d(self.client.request('GET', path='/hello'))
+        d.addCallback(json_content)
+
+        request = yield self.requests.get()
+        self.assertThat(request, HasRequestProperties(
+            method='GET', url=self.uri('/hello')))
+        self.assertThat(request.requestHeaders,
+                        HasHeader('accept', ['application/json']))
+
+        request.setResponseCode(200)
+        request.setHeader('Content-Type', 'application/json')
+        request.write(json.dumps({}).encode('utf-8'))
+        request.finish()
```

Comment:
We seem to be doing this sequence a lot. Is it something we can reasonably factor out into a helper function?